applications/ShapeOptimizationApplication/python_scripts/response_function_factory.py

```diff
@@ -23,35 +20,74 @@
 CheckForPreviousImport()
 
 # ==============================================================================
-def CreateSolver(model_part, opt_settings):
-
-    # Dictionary to store solvers of all response functions defined in the optimization settings
-    solver = {}
-
-    # Collect all responses
-    specified_responses = {}
-    for response_id in opt_settings.objectives:
-        specified_responses[response_id] = opt_settings.objectives[response_id]
-
-    if not specified_responses:
-        raise ValueError("No objective function specified!")
-
-    for response_id in opt_settings.constraints:
-        specified_responses[response_id] = opt_settings.constraints[response_id]
-
-    # Creat response function solver according to specified settings and add relevant variables
-    in_active_responses = True
-    if "strain_energy" in specified_responses.keys():
-        in_active_responses = False
-        model_part.AddNodalSolutionStepVariable(STRAIN_ENERGY_SHAPE_GRADIENT)
-        solver["strain_energy"] = StrainEnergyResponseFunction(model_part, specified_responses["strain_energy"])
-    if "mass" in specified_responses.keys():
-        in_active_responses = False
-        model_part.AddNodalSolutionStepVariable(MASS_SHAPE_GRADIENT)
-        solver["mass"] = MassResponseFunction(model_part, specified_responses["mass"])        
-    if in_active_responses:
-        raise ValueError("Specified response function not implemented. Implemented response functions are: \"strain_energy\", \"mass\"")
-
-    return solver
+def CreateListOfResponseFunctions( inputModelPart, optimizationSettings ):
+    listOfResponseFunctions = {}
+    responseCreator = ResponseFunctionCreator( inputModelPart, optimizationSettings )
+    responseCreator.AddSpecifiedKratosResponseFunctionsToList( listOfResponseFunctions )
+    return listOfResponseFunctions
+
+# ==============================================================================
+class ResponseFunctionCreator: 
+
+    # --------------------------------------------------------------------------
+    def __init__( self, inputModelPart, optimizationSettings ):
+        self.inputModelPart = inputModelPart
+        self.optimizationSettings = optimizationSettings
+
+     # --------------------------------------------------------------------------
+    def AddSpecifiedKratosResponseFunctionsToList( self, listOfResponseFunctions ):        
+        self.listOfResponseFunctions = listOfResponseFunctions
+        self.addObjectivesToListOfResponseFunctions()
+        self.addConstraintsToListOfResponseFunctions()
+        
+    # --------------------------------------------------------------------------
```

Comment:
Maybe you could think of underscores in the member function naming, to make it easier to see if a function is private.
Also i would not store "listOfResponseFunctions" as member of the class but pass it to the private functions